# Assignment 1: Summer olympics

**Notice I used PyCharm IDE and in PyCharm you don't need all the times the command print() to print the output. Also, if you don't used print() the output is a table you can play with.**

In [1]:
import pandas as pd
pd.__version__

'2.2.1'

In [3]:
summer = pd.read_csv('summer.csv')

In [60]:
summer

Year    City      Sport           Discipline  \
0      1896  Athens   Aquatics             Swimming   
1      1896  Athens   Aquatics             Swimming   
2      1896  Athens   Aquatics             Swimming   
3      1896  Athens   Aquatics             Swimming   
4      1896  Athens   Aquatics             Swimming   
...     ...     ...        ...                  ...   
31160  2012  London  Wrestling  Wrestling Freestyle   
31161  2012  London  Wrestling  Wrestling Freestyle   
31162  2012  London  Wrestling  Wrestling Freestyle   
31163  2012  London  Wrestling  Wrestling Freestyle   
31164  2012  London  Wrestling  Wrestling Freestyle   

                         Athlete Country Gender                       Event  \
0                  HAJOS, Alfred     HUN    Men              100M Freestyle   
1               HERSCHMANN, Otto     AUT    Men              100M Freestyle   
2              DRIVAS, Dimitrios     GRE    Men  100M Freestyle For Sailors   
3             MALOKINIS, Ioannis     GRE    Men  100M Freestyle For Sailors   
4             CHASAPIS, Spiridon     GRE    Men  100M Freestyle For Sailors   
...                          ...     ...    ...                         ...   
31160         JANIKOWSKI, Damian     POL    Men                    Wg 84 KG   
31161  REZAEI, Ghasem Gholamreza     IRI    Men                    Wg 96 KG   
31162             TOTROV, Rustam     RUS    Men                    Wg 96 KG   
31163          ALEKSANYAN, Artur     ARM    Men                    Wg 96 KG   
31164             LIDBERG, Jimmy     SWE    Men                    Wg 96 KG   

        Medal  
0        Gold  
1      Silver  
2      Bronze  
3        Gold  
4      Silver  
...       ...  
31160  Bronze  
31161    Gold  
31162  Silver  
31163  Bronze  
31164  Bronze  

[31165 rows x 9 columns]

In [61]:
summer.describe()

Year
count  31165.000000
mean    1970.482785
std       33.158416
min     1896.000000
25%     1948.000000
50%     1980.000000
75%     2000.000000
max     2012.000000

In [62]:
summer.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31165 entries, 0 to 31164
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Year        31165 non-null  int64 
 1   City        31165 non-null  object
 2   Sport       31165 non-null  object
 3   Discipline  31165 non-null  object
 4   Athlete     31165 non-null  object
 5   Country     31161 non-null  object
 6   Gender      31165 non-null  object
 7   Event       31165 non-null  object
 8   Medal       31165 non-null  object
dtypes: int64(1), object(8)
memory usage: 2.1+ MB


In [63]:
summer.Athlete

0                    HAJOS, Alfred
1                 HERSCHMANN, Otto
2                DRIVAS, Dimitrios
3               MALOKINIS, Ioannis
4               CHASAPIS, Spiridon
                   ...            
31160           JANIKOWSKI, Damian
31161    REZAEI, Ghasem Gholamreza
31162               TOTROV, Rustam
31163            ALEKSANYAN, Artur
31164               LIDBERG, Jimmy
Name: Athlete, Length: 31165, dtype: object

In [64]:
summer.shape

(31165, 9)

## Questions:

### Question 1:

- 1. Which country won the largest number of Bronze medals ever?

In [65]:
summer[summer.Medal == 'Bronze'].groupby(by =["Country"]).Medal.count().sort_values(ascending=False)

Country
USA    1098
URS     584
GBR     553
FRA     497
GER     475
       ... 
MKD       1
DJI       1
SYR       1
DOM       1
GUY       1
Name: Medal, Length: 123, dtype: int64

the answer is: USA and they won 1098 bronze medals over the years.

### Question 2:

- 2. How many athletes participated in the 2012 Olympics?

In [66]:
(summer[summer['Year']==2012].drop_duplicates(subset=['Athlete'], keep='first')
.Athlete.count())

1777

In [67]:
summer[summer['Year'] == 2012]['Athlete'].nunique()

1777

The number of athletes participate in 2012 Olympics is 1777 athlete

### Question 3:

- 3. List all the sports in which the United States has won at least one gold medal.

In [68]:
summer[(summer['Medal'] == 'Gold') & (summer['Country'] == 'USA')]['Event'].unique()

array(['100M', '110M Hurdles', '400M', 'Discus Throw', 'High Jump',
       'Long Jump', 'Pole Vault', 'Shot Put', 'Triple Jump',
       '25M Army Pistol', '50M Pistol (60 Shots)', '200M', '200M Hurdles',
       '400M Hurdles', '60M', 'Hammer Throw', 'High Jump, Standing',
       'Long Jump, Standing', 'Triple Jump, Standing', 'Individual',
       'Eight With Coxswain (8+)', '10M Platform', 'Plunge For Distance',
       '200M Freestyle', '400M Freestyle', '4X50Y Freestyle Relay',
       'Double American Round (60Y - 50Y - 40Y)',
       'Double Columbia Round (50Y - 40Y - 30Y)',
       'Double National Round (60Y - 50Y)',
       'Double York Round (100Y - 80Y - 60Y)', 'Team Round 60Y',
       'Teams Fita Round', '1500M', '3000M Steeplechase', '4Miles Team',
       '800M', 'Marathon', 'Triathlon (Long Jump / Shot Put / 100 Yards)',
       '- 47.63KG (Flyweight)', '+ 71.67KG (Heavyweight)',
       '47.63 - 52.16KG (Bantamweight)', '52.16 - 56.7KG (Featherweight)',
       '56.7 - 61.24KG (L

In [69]:
olympic_data = summer
print("This is all the sports USA won at least one gold medal")
olympic_data[(olympic_data['Country'] == 'USA') & (olympic_data['Medal'] == 'Gold')][['Sport', 'Event']].drop_duplicates().groupby('Sport')['Event'].value_counts()

This is all the sports USA won at least one gold medal


Sport      Event                        
Aquatics   100M Backstroke                  1
           100M Breaststroke                1
           100M Butterfly                   1
           100M Freestyle                   1
           10M Platform                     1
                                           ..
Wrestling  82 - 90KG (Light-Heavyweight)    1
           90 - 100KG (Heavyweight)         1
           97 - 130KG                       1
           Wf 74 KG                         1
           Wf 96 KG                         1
Name: count, Length: 250, dtype: int64

- Notice we refer to sports and also to event because me more interesting the event more than sport.

### Question 4:

- 4. Which male athlete has won the most medals in a single edition of the Olympics?

In [70]:
(summer[summer['Gender'] == 'Men'].groupby(by=['Year', 'Athlete']).Medal.count().sort_values(ascending=False))

Year  Athlete              
1980  DITYATIN, Aleksandr      8
2004  PHELPS, Michael          8
2008  PHELPS, Michael          8
1920  LEE, Willis              7
1972  SPITZ, Mark              7
                              ..
1956  ELDER, Henry James       1
      EINARSSON, Vilhjalmur    1
      EGRIBAS, Dursan Ali      1
      DZENELADZE, Roman        1
2012  ZOU, Shiming             1
Name: Medal, Length: 20463, dtype: int64

The two male athletes that won the most medals in single edition of the olympics is:
   1. Michael Phelps (In 2004 **and** in 2008)
   2. Aleksandr Dityatin (in 1980)     

### Question 5:

- 5. How many different countries have participated in the Olympics since 2000?

In [71]:
summer[summer['Year'] >= 2000]['Country'].nunique()

113

The number of countries participate in the Olympics from 2000 is: 113 countries.

### Question 6:

- 6. In which year did Russia win the most total medals, and how many did they win?

In [72]:
summer[(summer['Country'] == 'URS') | (summer['Country'] == 'RUS')].groupby(by=['Year']).Medal.count().sort_values(ascending=False)

Year
1980    442
1988    294
1976    285
1972    211
2004    192
1968    188
2000    188
1964    174
1956    169
1960    169
2008    143
2012    130
1952    117
1996    115
Name: Medal, dtype: int64

The year that Russia won the most medals is 1980 and they won 442 medals. 

In [73]:
summer[summer['Country'] == 'RUS'].groupby('Year').size().idxmax()

2004

### Question 7:

- 7. Identify the top 5 athletes with the most silver medals across all editions of the Olympics.

In [74]:
summer[summer['Medal'] == 'Silver'].groupby(by=['Athlete']).Medal.count().sort_values(ascending=False)

Athlete
VORONIN, Mikhail             6
DITYATIN, Aleksandr          6
BABASHOFF, Shirley Farber    6
GOROKHOVSKAYA, Mariya        5
VAN GRUNSVEN, Anky           5
                            ..
JOHNSON CEDENO, Yudel        1
JOHNSEN, Vibeke              1
JOHN, Sabine                 1
JOHIN                        1
AANING, Alf Lied             1
Name: Medal, Length: 9101, dtype: int64

The top 5 athletes with the most silver medals is:
   1. Mikhail Voronin (6 silver medals)
   2. Aleksandr Dityatin (6 silver medals)
   3. Shirley Farber Babashoff (6 silver medals)
   4. Mariya Gorokhovskaya (5 silver medals)
   5. Anky Van Grunsven (5 silver medals)
   
 - notice there are more than two athlete with 5 silver medals but this two was on the sort.

### Question 8

- 8. Which sport has the smallest number of different events?

In [75]:
summer.groupby('Sport')['Event'].nunique().sort_values()

Sport
Jeu de paume          1
Hockey                1
Handball              1
Softball              1
Football              1
Triathlon             1
Lacrosse              1
Polo                  1
Ice Hockey            1
Cricket               1
Roque                 1
Basque Pelota         1
Basketball            1
Baseball              1
Rugby                 1
Tug of War            1
Volleyball            2
Golf                  2
Badminton             2
Rackets               2
Modern Pentathlon     3
Skating               3
Table Tennis          3
Croquet               3
Water Motorsports     3
Equestrian            6
Tennis                6
Taekwondo             8
Fencing              11
Canoe                14
Canoe / Kayak        20
Rowing               22
Gymnastics           26
Archery              30
Cycling              38
Aquatics             46
Judo                 47
Weightlifting        55
Boxing               55
Sailing              56
Athletics            56
Shooting  

In [76]:
summer.groupby('Sport')['Event'].nunique().idxmin()

'Baseball'

There are 16 sports with only one event.

the first in my sort is "Jeu de paume".

### Question 9:

- 9. For each Olympics, how many female versus male athletes participated?

In [77]:
summer.groupby(by=['Year', 'Gender']).Athlete.nunique().unstack().fillna(0).astype(int)

Gender   Men  Women
Year               
1896      94      0
1900     407      7
1904     318      6
1908     724     14
1912     745     25
1920    1032     32
1924     729     39
1928     559     84
1932     493     50
1936     719     77
1948     646     81
1952     679     96
1956     669    104
1960     669    110
1964     753    150
1968     738    169
1972     870    183
1976     851    328
1980     874    383
1984     884    425
1988     925    446
1992    1022    515
1996    1015    694
2000    1049    800
2004    1020    812
2008    1024    842
2012     955    824

In [78]:
summer.groupby(['Year', 'Gender']).size()

Year  Gender
1896  Men        151
1900  Men        501
      Women       11
1904  Men        458
      Women       12
1908  Men        789
      Women       15
1912  Men        855
      Women       30
1920  Men       1255
      Women       43
1924  Men        833
      Women       51
1928  Men        615
      Women       95
1932  Men        555
      Women       60
1936  Men        791
      Women       84
1948  Men        718
      Women       96
1952  Men        750
      Women      139
1956  Men        739
      Women      146
1960  Men        737
      Women      145
1964  Men        820
      Women      190
1968  Men        819
      Women      212
1972  Men        947
      Women      238
1976  Men        924
      Women      381
1980  Men        958
      Women      429
1984  Men        973
      Women      486
1988  Men       1011
      Women      535
1992  Men       1105
      Women      600
1996  Men       1082
      Women      777
2000  Men       1126
      Women      889


In this table you can see the numbers of men participate in Olympics every year versus the women participate in the same Olympics.

### Question 10:

- 10. What is the total number of medals won by China in swimming?

In [79]:
summer[(summer['Country'] == 'CHN') & (summer['Discipline'] == 'Swimming')].Medal.count()

64

The total number of medals won by china is swimming is 64.

### Question 11

 - 11. Rank countries by the number of gold medals won in athletics historically.

In [80]:
summer[(summer['Sport'] == 'Athletics') & (summer['Medal'] == 'Gold')].groupby(by=['Country']).Medal.count().sort_values(ascending=False)

Country
USA    491
URS     79
GBR     67
FIN     60
GDR     50
      ... 
SLO      1
MOZ      1
ECU      1
NGR      1
CRO      1
Name: Medal, Length: 66, dtype: int64

This is the sorted list of the country how won the most gold medals.

### Question 12:

- 12. Which discipline had athletes from the most number of different countries in the latest Olympics?

In [81]:
summer[summer['Year'] == 2012].groupby(by=['Discipline']).Country.nunique().sort_values(ascending=False)

Discipline
Athletics                42
Wrestling Freestyle      29
Judo                     23
Shooting                 23
Taekwondo                21
Weightlifting            20
Boxing                   20
Rowing                   18
Swimming                 17
Canoe Sprint             16
Sailing                  15
Gymnastics Artistic      14
Fencing                  13
Cycling Track            11
Diving                    8
Cycling Road              8
Tennis                    8
Canoe Slalom              8
Badminton                 7
Modern Pentathlon         6
Water Polo                6
Mountain Bike             6
Archery                   6
Marathon swimming         6
Handball                  6
Football                  6
Jumping                   5
Table Tennis              5
Cycling BMX               5
Triathlon                 5
Volleyball                5
Basketball                5
Hockey                    5
Eventing                  4
Beach Volleyball          4
Gymnastic

That the list of discipline and the number of different countries how participate in them in the last Olympics in 2012.

### Question 13:

- 13. Find the athlete who improved the most in terms of medal color (e.g., from Bronze in one Olympics to Gold in the next) across successive Olympics.

check difference between two olympics:

In [82]:
olympic_years = summer['Year'].drop_duplicates().sort_values()

def is_four_year_difference(years):
    for i in range(len(years) - 1):  # Iterate up to the second to last element
        if years.iloc[i+1] - years.iloc[i] != 4:
            print("difference between " + str(years.iloc[i]) + " and " + str(years.iloc[i+1]) + " is " + str(years.iloc[i+1] - years.iloc[i]))

is_four_year_difference(olympic_years)
olympic_years

difference between 1912 and 1920 is 8
difference between 1936 and 1948 is 12


0        1896
151      1900
663      1904
1133     1908
1937     1912
2822     1920
4120     1924
5004     1928
5714     1932
6329     1936
7204     1948
8018     1952
8907     1956
9792     1960
10674    1964
11684    1968
12715    1972
13900    1976
15205    1980
16592    1984
18051    1988
19597    1992
21302    1996
23161    2000
25176    2004
27174    2008
29216    2012
Name: Year, dtype: int64

by notice to all Olympics are in 4 year difference I change my strategy.

Create new data frame of only the athlete that participate in two olympics in a row (The notes below from ChatGPT)

In [83]:
# Create an empty DataFrame to store the result
summer_consecutive_athlete = pd.DataFrame()

# Iterate over the sorted years
for i in range(len(olympic_years)-1):
    # Get the athletes who participated in two consecutive Olympics
    year1 = summer[summer['Year'] == olympic_years.iloc[i]]
    year2 = summer[summer['Year'] == olympic_years.iloc[i+1]]
    
    # Find the common athletes
    common_athletes = year1[year1['Athlete'].isin(year2['Athlete'])]
    
    # Append the common athletes to the result DataFrame
    summer_consecutive_athlete = pd.concat([summer_consecutive_athlete, common_athletes])

# Now 'summer_consecutive_athlete' contains the rows of athletes who participated in two consecutive Olympics, with data from the first Olympics

# To include data from the second Olympics, we need to iterate over the summer_consecutive_athlete DataFrame
for i in range(len(summer_consecutive_athlete)):
    athlete = summer_consecutive_athlete.iloc[i]['Athlete']
    year = summer_consecutive_athlete.iloc[i]['Year']
    
    # Find the athlete's data from the next Olympics
    next_year_data = summer[(summer['Year'] == olympic_years[olympic_years > year].min()) & (summer['Athlete'] == athlete)]
    
    # Check if the data already exists in the DataFrame
    if not summer_consecutive_athlete[(summer_consecutive_athlete['Athlete'] == athlete) & (summer_consecutive_athlete['Year'] == olympic_years[olympic_years > year].min())].empty:
        continue
    
    # Append the data to the result DataFrame
    summer_consecutive_athlete = pd.concat([summer_consecutive_athlete, next_year_data])

# Now 'summer_consecutive_athlete' contains the rows of athletes who participated in two consecutive Olympics, with data from both Olympics
summer_consecutive_athlete

Year    City      Sport           Discipline              Athlete  \
27     1896  Athens  Athletics            Athletics      GARRETT, Robert   
30     1896  Athens  Athletics            Athletics      CONNOLLY, James   
31     1896  Athens  Athletics            Athletics      GARRETT, Robert   
32     1896  Athens  Athletics            Athletics      CONNOLLY, James   
34     1896  Athens  Athletics            Athletics      GARRETT, Robert   
...     ...     ...        ...                  ...                  ...   
31135  2012  London  Wrestling  Wrestling Freestyle     GAZYUMOV, Khetag   
31136  2012  London  Wrestling  Wrestling Freestyle  GOGSHELIDZE, George   
31093  2012  London  Wrestling  Wrestling Freestyle      TAYMAZOV, Artur   
31142  2012  London  Wrestling  Wrestling Freestyle    BAYRAMOV, Rovshan   
31151  2012  London  Wrestling  Wrestling Freestyle       GUENOT, Steeve   

      Country Gender         Event   Medal  
27        USA    Men  Discus Throw    Gold  
30        USA    Men     High Jump  Silver  
31        USA    Men     High Jump  Silver  
32        USA    Men     Long Jump  Bronze  
34        USA    Men     Long Jump  Silver  
...       ...    ...           ...     ...  
31135     AZE    Men      Wf 96 KG  Bronze  
31136     GEO    Men      Wf 96 KG  Bronze  
31093     UZB    Men      Wf 120KG    Gold  
31142     AZE    Men      Wg 55 KG  Silver  
31151     FRA    Men      Wg 66 KG  Bronze  

[9455 rows x 9 columns]

Checking codes to see the output is valid:

In [84]:
summer_consecutive_athlete.groupby(by=['Athlete']).Year.value_counts()

Athlete                   Year
AALTONEN, Paavo Johannes  1948    4
                          1952    1
ABALO, Luc                2008    1
                          2012    1
ABBAGNALE, Agostino       1996    1
                                 ..
ZYKINA, Olesya            2004    1
ZÜRNER, Albert            1908    1
                          1912    1
ÖSTENSEN, Östen           1920    4
                          1912    1
Name: count, Length: 7867, dtype: int64

In [85]:
summer_consecutive_athlete[summer_consecutive_athlete['Athlete'] == 'AALTONEN, Paavo Johannes']

Year      City       Sport   Discipline                   Athlete  \
7709  1948    London  Gymnastics  Artistic G.  AALTONEN, Paavo Johannes   
7716  1948    London  Gymnastics  Artistic G.  AALTONEN, Paavo Johannes   
7730  1948    London  Gymnastics  Artistic G.  AALTONEN, Paavo Johannes   
7773  1948    London  Gymnastics  Artistic G.  AALTONEN, Paavo Johannes   
8563  1952  Helsinki  Gymnastics  Artistic G.  AALTONEN, Paavo Johannes   

     Country Gender                 Event   Medal  
7709     FIN    Men  Individual All-Round  Bronze  
7716     FIN    Men          Pommel Horse    Gold  
7730     FIN    Men      Team Competition    Gold  
7773     FIN    Men                 Vault    Gold  
8563     FIN    Men      Team Competition  Bronze

Create new column of the medals values:

In [86]:
def medals_values (row):
    if row['Medal'] == 'Gold':
        return 3
    elif row['Medal'] == 'Silver':
        return 2
    else:
        return 1
summer_consecutive_athlete['Medals_values'] = summer_consecutive_athlete.apply(medals_values, axis=1)
summer_consecutive_athlete

Year    City      Sport           Discipline              Athlete  \
27     1896  Athens  Athletics            Athletics      GARRETT, Robert   
30     1896  Athens  Athletics            Athletics      CONNOLLY, James   
31     1896  Athens  Athletics            Athletics      GARRETT, Robert   
32     1896  Athens  Athletics            Athletics      CONNOLLY, James   
34     1896  Athens  Athletics            Athletics      GARRETT, Robert   
...     ...     ...        ...                  ...                  ...   
31135  2012  London  Wrestling  Wrestling Freestyle     GAZYUMOV, Khetag   
31136  2012  London  Wrestling  Wrestling Freestyle  GOGSHELIDZE, George   
31093  2012  London  Wrestling  Wrestling Freestyle      TAYMAZOV, Artur   
31142  2012  London  Wrestling  Wrestling Freestyle    BAYRAMOV, Rovshan   
31151  2012  London  Wrestling  Wrestling Freestyle       GUENOT, Steeve   

      Country Gender         Event   Medal  Medals_values  
27        USA    Men  Discus Throw    Gold              3  
30        USA    Men     High Jump  Silver              2  
31        USA    Men     High Jump  Silver              2  
32        USA    Men     Long Jump  Bronze              1  
34        USA    Men     Long Jump  Silver              2  
...       ...    ...           ...     ...            ...  
31135     AZE    Men      Wf 96 KG  Bronze              1  
31136     GEO    Men      Wf 96 KG  Bronze              1  
31093     UZB    Men      Wf 120KG    Gold              3  
31142     AZE    Men      Wg 55 KG  Silver              2  
31151     FRA    Men      Wg 66 KG  Bronze              1  

[9455 rows x 10 columns]

check who is the most improved athletic:

In [87]:
result = summer_consecutive_athlete.groupby(by= ['Athlete', 'Year']).Medals_values.sum().groupby(level=0).diff().sort_values(ascending=False)
result

Athlete              Year
BIONDI, Matthew      1988    15.0
DITYATIN, Aleksandr  1980    14.0
SPITZ, Mark          1972    12.0
NADI, Nedo           1920    12.0
ANDRIANOV, Nikolay   1976    11.0
                             ... 
ZWOLLE, Hendrik Jan  1992     NaN
ZYBINA, Galina       1952     NaN
ZYKINA, Olesya       2000     NaN
ZÜRNER, Albert       1908     NaN
ÖSTENSEN, Östen      1912     NaN
Name: Medals_values, Length: 7867, dtype: float64

The athlete how improve the most between two consecutive Olympics is Matthew Biondi in from 1984 Olympics to 1988 Olympics.

 ### Question 14:

- 14. What is the average number of participating countries in each Olympic sport?

In [88]:
summer.groupby(['Sport', 'Year'])['Country'].nunique().groupby(['Sport']).mean()

Sport
Aquatics             13.370370
Archery               4.800000
Athletics            22.629630
Badminton             5.333333
Baseball              3.000000
Basketball            3.888889
Basque Pelota         2.000000
Boxing               16.250000
Canoe                19.000000
Canoe / Kayak        12.000000
Cricket               2.000000
Croquet               1.000000
Cycling               9.222222
Equestrian            7.375000
Fencing               7.407407
Football              3.560000
Golf                  2.000000
Gymnastics            8.037037
Handball              4.916667
Hockey                3.636364
Ice Hockey            3.000000
Jeu de paume          2.000000
Judo                 17.250000
Lacrosse              2.000000
Modern Pentathlon     3.347826
Polo                  2.200000
Rackets               1.000000
Roque                 1.000000
Rowing               11.807692
Rugby                 2.500000
Sailing              10.560000
Shooting             12.440000
Sk

In [89]:
summer.groupby('Sport')['Country'].nunique().mean()

25.372093023255815

That a list of all the mean participation countries in each Olympics in each Olympics sport.

### Question 15:

- 15. Compare the total number of medals won by Germany before and after the reunification in 1990.

number of medals for germany before the reunification:

In [90]:
summer[(summer['Year'] < 1990) &
       ((summer['Country'] == 'GER') |
       (summer['Country'] == 'EUA') |
       (summer['Country'] == 'FRG') |
       (summer['Country'] == 'GDR')) ].Medal.value_counts().sum()

2095

number of medals for germany after the reunification:

In [91]:
summer[(summer['Year'] >= 1990) &
       (summer['Country'] == 'GER')].Medal.value_counts().sum()

785

The number of medals germany won before the reunification in 1990 is 2095 she used the next NOCs:
1. GER as Germany
2. FRG as West Germany
3. GDR as East Germany
4. EUA as United Team of Germany

The number of medals germany won after the reunification in 1990 is 785 she used the next NOC:
1. GER as Germany

In [96]:
before_1990 = summer[(summer['Country'] == 'GER') & (summer['Year'] < 1990)].shape[0]
after_1990 = summer[(summer['Country'] == 'GER') & (summer['Year'] >= 1990)].shape[0]
before_1990

520

In [93]:
before_1990

520

In [94]:
after_1990

785